models/cross-spork_client.go

```diff
@@ -0,0 +1,142 @@
+package models
+
+import (
+	"context"
+	"fmt"
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/access"
+	"github.com/rs/zerolog"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+// CrossSporkClient is a wrapper around the Flow AN client that can
+// access different AN APIs based on the height boundaries of the sporks.
+//
+// Each spork is defined with the last height included in that spork,
+// based on the list we know which AN client to use when requesting the data.
+//
+// Any API that supports cross-spork access must have a defined function
+// that shadows the original access Client function.
+type CrossSporkClient struct {
+	logger zerolog.Logger
+	// this map holds the last heights and clients for each spork
+	sporkHosts map[uint64]access.Client
+	access.Client
+}
+
+// NewCrossSporkClient creates a new instance of the client, it accepts the
+// host to the current spork AN API.
+func NewCrossSporkClient(currentSporkClient access.Client, logger zerolog.Logger) (*CrossSporkClient, error) {
```

Comment:
There isn't any statement in this methods' body that can cause an error. If we choose to move the instantiation of current spork client and past sport clients in this method, then it makes sense to keep it.